File: src/org/kde/kdeconnect/UserInterface/List/PluginItem.java
/*
 * SPDX-FileCopyrightText: 2014 Albert Vaca Cintora <albertvaka@gmail.com>
 *
 * SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
*/

package org.kde.kdeconnect.UserInterface.List;

import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;

import org.kde.kdeconnect.Plugins.Plugin;
import org.kde.kdeconnect_tp.R;
import org.kde.kdeconnect_tp.databinding.ListItemWithIconEntryBinding;

public class PluginItem extends EntryItemWithIcon {
    private final View.OnClickListener clickListener;

    public PluginItem(Plugin p, View.OnClickListener clickListener) {
        super(p.getActionName(), p.getIcon());
        this.clickListener = clickListener;
    }

    @NonNull
    @Override
    public View inflateView(@NonNull LayoutInflater layoutInflater) {
        final View root = super.inflateView(layoutInflater);
        root.setOnClickListener(clickListener);
        return root;
    }
}

File: src/org/kde/kdeconnect/UserInterface/List/PluginItem.kt
package org.kde.kdeconnect.UserInterface.List

import android.content.Context
import android.graphics.drawable.Drawable
import org.kde.kdeconnect.Plugins.Plugin

class PluginItem(
    val context: Context,
    val header: String,
    val textStyleRes: Int? = null,
) {

    var action: (() -> Unit)? = null
    var icon: Drawable? = null

    constructor(
        context: Context,
        plugin: Plugin,
        action: (Plugin) -> Unit,
        textStyleRes: Int? = null,
    ) : this(
        context = context,
        header = plugin.displayName,
        textStyleRes = textStyleRes,
    ) {
        this.action = { action(plugin) }
        this.icon = plugin.icon
    }
}